test_app/authentication/service_token_auth.py

```diff
@@ -0,0 +1,39 @@
+import jwt
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from rest_framework.authentication import BaseAuthentication, get_authorization_header
+
+from ansible_base.resource_registry.resource_server import get_resource_server_config
+
+User = get_user_model()
+
+
+class ServiceTokenAuthentication(BaseAuthentication):
+    keyword = "Token"
+
+    def authenticate(self, request):
+        auth = get_authorization_header(request).split()
+
+        if not auth or auth[0].lower() != self.keyword.lower().encode():
+            return None
+
+        token = auth[1]
+
+        cfg = get_resource_server_config()
```

Comment:
I'm not sure if I'm in love with this config. But maybe I am seeing why you are using it. Its like another layer on top of settings. I guess it makes it so that we don't have to do `settings.get('JWT_ALGORYTHM', <default value>)` everywhere but I fell like we could just do something like this in the dynamic settings or somewhere "more traditional":
```
if 'JWT_ALGORYTHM' not in <whatever settings data structure>:
    <whatever settings data structure>['JWT_ALGORYTHM'] = 'RSA256'
```

My concern with the config is people might not know about it and then end up just referencing the setting directly anyway. If we made these configs a larger pattern I suppose that would work as well but it kind of feels like a one off here.

Reply:
I think it makes sense because it makes it a lot easier to set defaults, you can take advantage of typing and you don't run as much risk of settings colliding. I do see your point about accessing it directly though. Because this is a dict in settings.py, I can't just add it to settings as a default because redefining it would remove the default keys.

Reply:
I think I would prefer to keep this pattern. It seems more sane than guessing the default value everywhere that `settings.RESOURCE_SERVER` is referenced.